src/GraphQL/Utilities/Visitors/SchemaValidationVisitor.cs

```diff
@@ -0,0 +1,186 @@
+using System;
+using System.Linq;
+using GraphQL.Types;
+
+namespace GraphQL.Utilities
+{
+    internal sealed class SchemaValidationVisitor : BaseSchemaNodeVisitor
```

Comment:
Many checks in this class are not achievable in practice (for example names starting with "__"), because we try to control the correctness of the schema definition as early as possible. Nevertheless, I'm sure it would be correct to have in this class the full set of checks required by the specification. Perhaps more will be added over time. Probably not if we go to have this class reflect standard spec requirements. Other checks can always be placed in other visitors.

Reply:
Even if sealed, it should still be public so that a custom schema implementation can call it.

Reply:
done